src/convolution.jl

```diff
@@ -47,6 +50,20 @@ end
 convolve(d1::Poisson, d2::Poisson) =  Poisson(d1.λ + d2.λ)
 
 
+
+function convolve(d1::DiscreteNonParametric, d2::DiscreteNonParametric)
+    support_conv = vcat([d1.support .+ v for v in d2.support]...) |> unique 
+    sort!(support_conv) #for fast index finding below
+    p_conv = zeros(length(support_conv)) 
```

Comment:
This should not necessarily be a vector of `Float64`. Maybe
```suggestion
    probs1 = probs(d1)
    probs2 = probs(d2)
    p_conv = zeros(Base.promote_eltype(probs1, probs2), length(support_conv)) 
```
even though this would also not preserve the array type, e.g. if they would be static arrays.